src/java.desktop/unix/classes/sun/awt/wl/WLToolkit.java

```diff
@@ -955,7 +955,8 @@ protected boolean needsXEmbedImpl() {
     @Override
     public boolean isModalityTypeSupported(Dialog.ModalityType modalityType) {
         return (modalityType == Dialog.ModalityType.MODELESS) ||
-                (modalityType == Dialog.ModalityType.APPLICATION_MODAL);
+                (modalityType == Dialog.ModalityType.APPLICATION_MODAL) ||
+                (modalityType == Dialog.ModalityType.DOCUMENT_MODAL);
```

Comment:
Not sure DOCUMENT_MODAL is naturally supported by the rest of the toolkit. Could you please point to some tests that show that this is actually supported according to the documentation:
> A DOCUMENT_MODAL dialog blocks input to all top-level windows from the same document except those from its own child hierarchy.

Reply:
Within the real-world JOSM use case and KDE Wayland 6.5.4, all the parent windows of the DOCUMENT_MODAL dialog don't receive mouse click events (but window focusing events still work which isn't the case with XWayland and X11 driver)

I guess I can write more tests confirming this behavior if needed (and try to fix the window focus side-effect)